source/fuzz/fuzzer_pass_add_loads.cpp

```diff
@@ -79,13 +83,46 @@ void FuzzerPassAddLoads::Apply() {
           return;
         }
 
-        // Choose a pointer at random, and create and apply a loading
-        // transformation based on it.
-        ApplyTransformation(TransformationLoad(
-            GetFuzzerContext()->GetFreshId(),
+        auto chosen_instruction =
             relevant_instructions[GetFuzzerContext()->RandomIndex(
-                                      relevant_instructions)]
-                ->result_id(),
+                relevant_instructions)];
+
+        bool is_atomic_load = false;
+        uint32_t memory_scope_id = 0;
+        uint32_t memory_semtantics_id = 0;
+
+        // has work group or storage buffer (storage class)
+        if (chosen_instruction->GetInOperand(0).words[0] ==
+                SpvMemorySemanticsUniformMemoryMask ||
+            chosen_instruction->GetInOperand(0).words[0] ==
+                SpvMemorySemanticsWorkgroupMemoryMask) {
```

Comment:
Done.
Regarding the `FuzzerPassAddLoads::GetAtomicMemorySemanticsMaskFromType(uint32_t target_pointer_instruction)` I will be planing for this because I will like to support unit tests for all current fuzzer or at least I have made (but in separate PR).
What is your opinion @paulthomson?